src/ratchet.rs

```diff
@@ -146,14 +160,32 @@ impl Ratchet {
     /// let mut ratchet = Ratchet::new(&mut rand::thread_rng());
     /// ratchet.inc_by(3);
     ///
-    /// println!("{:?}", String::from(&ratchet));
+    /// println!("{:?}", ratchet);
     /// ```
     pub fn inc_by(&mut self, n: usize) {
-        let (jumped, _) = inc_by(self, n);
-        *self = jumped;
+        if n == 0 {
+            return;
+        } else if n >= LARGE_EPOCH_LENGTH - self.combined_counter() {
+            // `n` is larger than at least one large epoch jump
+            let jump_count = self.next_large_epoch();
+            self.inc_by(n - jump_count);
+            return;
+        } else if n >= MEDIUM_EPOCH_LENGTH - self.small_counter as usize {
+            // `n` is larger than at least one medium epoch jump
+            let jump_count = self.next_medium_epoch();
+            self.inc_by(n - jump_count);
+            return;
+        }
+
+        self.small = Hash::from_chain([], self.small, n);
```

Comment:
Aha! Here's the `from_chain` that I asked about above! It makes sense in context 👍 `from_chain` is not a super intuitive name for me, but I'm guessing it's related to the SHA3 crate, which has a `chain_update` method.